### PR TITLE
Feature #87 - Add the reset option to the different combobox

### PR DESCRIPTION
**Pull request by rochiebcn**
Implemented the new functionality for the comboboxes that allow to put a X to remove the option selected

### CODE REVIEW

File: src/app/systelab-components/select/gender-combobox.component.ts [hunk 1]
```diff
@@ -55,4 +60,17 @@ export class GenderSelect extends AbstractComboBox<Element> implements AfterView
 	getIdField(): string {
 		return 'id';
 	}
+
+
+	deleteValueSelected(event: MouseEvent): void {
+		event.stopPropagation();
+		if (this.showAll) {
+			this._id = 'A';
```
Comment (alfonsserra) [inline src/app/systelab-components/select/gender-combobox.component.ts]: As you have and id, then we are going to have the close icon always. as the cross is visible when "withDeleteOption && id"

File: src/app/systelab-components/select/gender-combobox.component.ts [hunk 2]
```diff
@@ -55,4 +60,17 @@ export class GenderSelect extends AbstractComboBox<Element> implements AfterView
...
+			this.description = this._descriptionAll;
+		} else {
+			this._id = 'U';
+			this.description = this._descriptionUnknown;
```
Comment (alfonsserra) [inline src/app/systelab-components/select/gender-combobox.component.ts]: To be coherent, and as the previous one:

why don't you do:
this._description = this._descriptionUnknown;
this.id = 'U';

File: src/app/systelab-components/select/all-yes-no-combobox.component.ts [hunk 3]
```diff
@@ -14,8 +14,11 @@ class Element {
 
 export class AllYesNoSelect extends AbstractComboBox<Element> implements AfterViewInit {
 
+	_descriptionAll: string;
```
Comment (alfonsserra) [inline src/app/systelab-components/select/all-yes-no-combobox.component.ts]: set as private

Comment (alfonsserra) [inline src/app/systelab-components/select/all-yes-no-combobox.component.ts]: Do not use the _ prefix. Is only for private and when you define set and get.

File: src/app/systelab-components/select/gender-combobox.component.ts [hunk 4]
```diff
@@ -16,17 +16,22 @@ class Element {
 export class GenderSelect extends AbstractComboBox<Element> implements AfterViewInit {
 
 	@Input() showAll = false;
+	private _descriptionAll;
```
Comment (alfonsserra) [inline src/app/systelab-components/select/gender-combobox.component.ts]: Do not use the _ prefix. Is only for private and when you define set and get.

File: src/app/systelab-components/select/gender-combobox.component.ts [hunk 5]
```diff
@@ -16,17 +16,22 @@ class Element {
...
+	private _descriptionUnknown;
```
Comment (alfonsserra) [inline src/app/systelab-components/select/gender-combobox.component.ts]: Do not use the _ prefix. Is only for private and when you define set and get.

**Comment by rochiebcn** [comment]
- Has been changed the funcionality of deleteValueSelected in order to make that is not needed to override for add the default value in combos without the blank option.
- Has been added to the abstract-combobox component the inputs defaultIdValue, defaultDescription and defaultCode to have the possibility of define a item by default. This values will override the undefined values when the delete option of the combo will be performed

**Comment by rochiebcn** [comment]
I don't think that it's neccesary to use this functionality in the autocomplete combobox because it's a editable component.
As the tree and api combobox is a child of abstract-combobox, it's possible to use the functionality

File: src/app/systelab-components/select/gender-combobox.component.ts (hunk 5 above)
Comment (rochiebcn) [reply src/app/systelab-components/select/gender-combobox.component.ts]: Ok. Fix it

**Pull request merged**